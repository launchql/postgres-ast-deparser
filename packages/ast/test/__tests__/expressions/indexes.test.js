import { getConnections } from '../../utils';
import { indexes } from './__fixtures__/indexes';

let db, teardown;
const objs = {
  tables: {}
};

beforeAll(async () => {
  ({ db, teardown } = await getConnections());
  await db.begin();
  await db.savepoint();
});
afterAll(async () => {
  try {
    //try catch here allows us to see the sql parsing issues!
    await db.rollback();
    await db.commit();
    await teardown();
  } catch (e) {}
});

it('index_stmt', async () => {
    for (const index of indexes) {
      const [{ deparse: result }] = await db.any(
        `
  select deparser.deparse(
    $1::jsonb
  );
    `,
        [index]
      );
      expect(result).toMatchSnapshot();
    }
  });