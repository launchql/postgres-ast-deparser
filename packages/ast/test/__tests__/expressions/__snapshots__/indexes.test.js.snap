// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index_stmt 1`] = `"CREATE INDEX idx_simple ON table_simple (simple);"`;

exports[`index_stmt 2`] = `"CREATE INDEX idx_composite ON table_composite (col1, col2);"`;

exports[`index_stmt 3`] = `"CREATE INDEX idx_include ON table_include (col1) INCLUDE (col2, col3);"`;

exports[`index_stmt 4`] = `"CREATE INDEX \\"idx_CaseSimple\\" ON \\"Table_CaseSimple\\" (\\"CaseSimple\\");"`;

exports[`index_stmt 5`] = `"CREATE INDEX \\"idx_CaseInclude\\" ON table_caseinclude (simple) INCLUDE (\\"CaseInclude\\");"`;
